File: musicShopProject/Model/Categories/CategoryBlank.cs
namespace musicShopProject.Model.Categories;

public partial class CategoryBlank
{
    public Guid? Id { get; set; }

    public String? Name { get; set; }

    public String? Photo { get; set; }

    public Guid? CreatedUserId { get; set; }

    public Guid? ModifiedUserId { get; set; }
}

public partial class CategoryBlank
{
    public class Validated
    {
        public Guid Id { get; }

        public String Name { get; }

        public String Photo { get; }

        public Guid CreatedUserId { get; }

        public Guid? ModifiedUserId { get; }

        public Validated(Guid id, String name, String photo, Guid createdUserId, Guid? modifiedUserId)
        {
            Id = id;
            Name = name;
            Photo = photo;
            CreatedUserId = createdUserId;
            ModifiedUserId = modifiedUserId;
        }
    }
}

File: musicShopProject/ClientApp/src/domain/products/productsCategories/categoryBlank.ts
import { Category } from "./category";

export interface CategoryBlank{
    id: string | null,
    name: string | null,
    photo: string | null
}

export namespace CategoryBlank {
    export function getEmpty(): CategoryBlank {
        return {
            id: null,
            name: null,
            photo: null
        };
    }

    export function toProductBlank(category: Category): CategoryBlank {
        return {
            id: category.id,
            name: category.name,
            photo: category.photo
        };
    }
 }